python/plot_hist_sgp4Scan.py

```diff
@@ -0,0 +1,178 @@
+print ""
+
+# Start timer.
+start_time = time.time( )
+
+print ""
+print "******************************************************************"
+print "                          Input parameters                        "
+print "******************************************************************"
+print ""
+
+# Parse JSON configuration file
+# Raise exception if wrong number of inputs are provided to script
+if len(sys.argv) != 2:
+    raise Exception("Only provide a JSON config file as input!")
+
+json_data = open(sys.argv[1])
+config = commentjson.load(json_data)
+json_data.close()
+pprint(config)
+
+print ""
+print "******************************************************************"
+print "                            Operations                            "
+print "******************************************************************"
+print ""
+
+print "Input data files being read ..."
+
+output_path_prefix = config["output_directory"] + '/'
+
+print "Fetching scan data from database ..."
+
+# Connect to SQLite database.
+try:
+    database = sqlite3.connect(config['database'])
+
+except sqlite3.Error, e:
+    print "Error %s:" % e.args[0]
+    sys.exit(1)
+
+# Fetch scan data.
+scan_data = pd.read_sql( "SELECT    " + config['error'] + "_x_error, \
+                                    " + config['error'] + "_y_error, \
+                                    " + config['error'] + "_z_error, \
+                                    " + config['error'] + "_error    \
+                          FROM      sgp4_scanner_results             \
+                          WHERE     success = 1;",
+                          database )
+
+scan_data.columns = [ 'xerror', 'yerror', 'zerror', 'magnitudeError' ]
+
+# The histogram of the data
+xerror = scan_data[ 'xerror' ]
+yerror = scan_data[ 'yerror' ]
+zerror = scan_data[ 'zerror' ]
+magnitudeError = scan_data[ 'magnitudeError' ]
+
+# Calculate mean, variance and standard deviation
+mu = sum( magnitudeError ) / len( magnitudeError )
+print 'Mean = ' + repr( mu )
+
+sumOfSquareDeviations = sum( ( x - mu )**2 for x in magnitudeError )
+variance = sumOfSquareDeviations / len( magnitudeError )
+print 'Variance = ' + repr( variance )
+
+sigma = variance**0.5
+print 'Standard Deviation = ' + repr( sigma )
+
+# Plot the magnitude of the error
+n, bins, patches = plt.hist( magnitudeError, bins=100, normed=True, facecolor='green', alpha=0.5,
+                             label='Magnitude' )
+
+# Add a line of expected distribution
+# y = mlab.normpdf( bins, mu, sigma )
+# l = plt.plot( bins, y, 'r--', linewidth=1.5, label='Best fit line' )
+
+# Select appropriate unit and title for the error type
+if config['error'] == 'arrival_position':
+    errorUnit = "[km]"
+    plotTitle = 'Arrival Position Error'
+else:
+    errorUnit = "[km/s]"
+    plotTitle = 'Arrival Velocity Error'
+
+# Figure properties
+plt.xlabel( 'Error' + " " + errorUnit )
+plt.ylabel( 'Normed Frequency' )
+plt.title( plotTitle + " " + 'Magnitude' )
```

Comment:
For papers we don't want to produce titles, maybe add a config option for this as well